frontend/src/components/views/customer_view/CustomerOrderFood.jsx

```diff
@@ -6,24 +6,82 @@ import Axios from 'axios';
 import Cart from './Cart';
 import config from '../../../config.json';
 import customerCartContext from './customerCartContext';
+import RestaurantCard from './RestaurantCard';
+
 
 class CustomerOrderFood extends Component {
   constructor(props) {
     super(props);
     this.state = {
       listOfRestaurants: [],
-      shoppingCart: ['1', '2', '3'],
+      shoppingCart: [],
     };
 
     this.loadRestaurants = async () => {
       const result = await Axios.get(`${config.localhost}restaurants/`);
       if (result.status === 200) {
-        console.log(result.data.restaurants);
         this.setState({ listOfRestaurants: result.data.restaurants });
       } else {
         alert('cannot load restaurant');
       }
     };
+    this.addToCart = (food) => {
+      const { shoppingCart } = this.state;
+      shoppingCart.push(food);
+      this.setState({ shoppingCart });
+    };
+
+    this.removeFromCart = (index) => {
+      let { shoppingCart } = this.state;
+      shoppingCart = shoppingCart.filter((x, current) => current !== index);
+      this.setState({ shoppingCart });
+    };
+    this.clearCart = () => {
+      this.setState({ shoppingCart: [] });
+    };
+    this.calculateTotal = () => {
+      const { shoppingCart } = this.state;
+      const priceList = shoppingCart.map((x) => Number(x.price));
+      const result = priceList.reduce((prev, curr) => prev + curr, 0);
+      console.log(result);
+      return result;
+    };
+    this.createOrder = async () => {
+      const { shoppingCart } = this.state;
+      const { customer_id } = this.props.user;
```

Comment:
Should this be obtained from userContext?

Reply:
why not Oo